modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/quantity_selector/QuantitySelector.js

```diff
@@ -12,246 +12,106 @@
  * details.
  */
 
+import {ClaySelect} from '@clayui/form';
 import ClayIcon, {ClayIconSpriteContext} from '@clayui/icon';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
-import React, {createRef, useEffect, useState} from 'react';
+import React, {useEffect, useState} from 'react';
 
-import throttle from '../../utilities/throttle';
-
-const THROTTLE_TIMEOUT = 1000;
+import Datalist from '../datalist/Datalist';
+import {generateOptions, getInputStyle} from './utils/index';
 
 function QuantitySelector(props) {
-	const [currentQuantity, setCurrentQuantity] = useState(
-		!!props.minQuantity && props.minQuantity > props.quantity
-			? props.minQuantity
-			: props.quantity
-	);
-	const [nextAvailable, setNextAvailable] = useState(
-		currentQuantity + props.multipleQuantity <= props.maxQuantity
-	);
-	const [prevAvailable, setPrevAvailable] = useState(
-		currentQuantity - props.multipleQuantity >= props.minQuantity
-	);
-	const [isThrottling, setIsThrottling] = useState(false);
-
-	const inputRef = createRef();
-
-	useEffect(() => {
-		setCurrentQuantity(props.quantity);
-	}, [props.quantity, setCurrentQuantity]);
-
-	useEffect(() => {
-		if (props.throttleOnUpdate) {
-			setIsThrottling(true);
-
-			props.onUpdate(currentQuantity).then(() => setIsThrottling(false));
-		}
-		else {
-			props.onUpdate(currentQuantity);
-		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [currentQuantity]);
-
-	useEffect(() => {
-		setNextAvailable(
-			currentQuantity + props.multipleQuantity <= props.maxQuantity
-		);
-		setPrevAvailable(
-			currentQuantity - props.multipleQuantity >= props.minQuantity
-		);
-	}, [
-		currentQuantity,
-		props.maxQuantity,
-		props.minQuantity,
-		props.multipleQuantity,
-	]);
-
-	function updateCurrentQuantity(newQuantity) {
-		if (
-			newQuantity >= props.minQuantity &&
-			newQuantity <= props.maxQuantity &&
-			newQuantity % props.multipleQuantity === 0
-		) {
-			setCurrentQuantity(newQuantity);
-		}
-	}
-
-	function _increaseQuantity() {
-		if (nextAvailable) {
-			updateCurrentQuantity(currentQuantity + props.multipleQuantity);
+	const [selectedQuantity, setSelectedQuantity] = useState(props.quantity);
+	const quantitiesList = generateOptions(props.settings);
+	const onUpdate = (quantity) => {
+		const qty = parseInt(quantity, 10);
+		if (qty) {
+			setSelectedQuantity(qty);
+			try {
+				props.updateQuantity('select', qty);
+			}
+			catch (event) {
+				// eslint-disable-next-line no-console
+				console.log('stand alone component');
```

Comment:
If you really need to use `console.log`'s, being inside a catch, please make sure it's for a good reason. The `try/catch` might blow up and the poor dev will never know why. :/